src/actions/write/tick.ts

```diff
@@ -120,20 +119,19 @@ function tickReservedNames({
   return activeReservedNames;
 }
 
-function tickGatewayRegistry({
+export function tickGatewayRegistry({
   currentBlockHeight,
   gateways,
   balances,
-  registrySettings,
 }: {
   currentBlockHeight: BlockHeight;
   gateways: DeepReadonly<Record<string, Gateway>>;
   balances: DeepReadonly<Record<string, number>>;
-  registrySettings: DeepReadonly<GatewayRegistrySettings>;
 }): {
   gateways: Record<string, Gateway>;
   balances: Record<string, number>;
 } {
+  // TODO: don't clone it until you need to
```

Comment:
Could create a new object of balance updates and then once it's done being constructed, do something like:
```
const updatedBalances = Object.keys(balanceUpdates).length ? { ...balances, ...balanceUpdates } : balances as Balances;
```
where you use `as` to cast away the DeepReadOnly-ness and save yourself some copying.

Reply:
optimized in [7a71932](https://github.com/ar-io/testnet-contract/pull/110/commits/7a71932d160954116a800a3399f8fa593088767e)